hxrsnd/macromotor.py

```diff
@@ -184,8 +193,137 @@ def set(self, position, wait=True, verify_move=True, ret_status=True,
             # number of status objects
             return self._status[0]
 
+    def configure(self, *, calib=None):
+        """
+        Configure the macro-motor's move parameters.
+
+        Parameters
+        ----------
+        calib : DataFrame or dict, optional
+            Lookup table for move calibration. This represents set positions of
+            auxiliary movers that should be chosen as we move our main macro.
+            If a DataFrame, the index should be the primary axis of motion, the
+            argument to the move or set functions. The columns should be the
+            desired set positions of each auxiliary mover at each index point.
+            The columns should be labeled with the name of the mover. The names
+            are interpreted as attribute access of the parent object, e.g.
+            name=t1 ------> snd.t1
+            name=t1.chi1 -> snd.chi1
+            If a dict, I'm either expecting the arugments to create a dataframe
+            e.g. data=dict(name=array, name2=array2), index=array3
+            OR a dictionary of names to functions of single variables, taking
+            in the main axis position and outputing the mover's adjustment.
+
+        Returns
+        -------
+        configs : tuple of dict
+            old_config, new_config
+        """
+        # Save prev for return statement
+        prev_config = self.read_configuration()
+        self._config_calib(calib)
+        return prev_config, self.read_configuration()
+
+    def _config_calib(self, calib):
+        """
+        Handle calib arg from configure
+        """
+        # Interpret calib
+        if calib is None:
+            save_calib = {}
+        elif isinstance(calib, pd.DataFrame):
+            save_calib = calib
+        elif isinstance(calib, dict):
+            try:
+                save_calib = pd.Dataframe(**calib)
+            except Exception:
+                save_calib = calib
+        else:
+            raise TypeError("Invalid calib type {}".format(type(calib)))
+
+        # Check for valid inputs
+        if isinstance(save_calib, pd.DataFrame):
+            names = save_calib.columns
+        elif isinstance(save_calib, dict):
+            for name, func in save_calib.items():
+                if not callable(func):
+                    err = 'Recieved non-callable for {}'.format(name)
+                    raise TypeError(err)
+            names = save_calib.keys()
```

Comment:
Roll these into a single `if`, `elif` block for readability or am I missing something?

Reply:
Not 100% sure what you mean, the blocks above and below `# Check for valid inputs` need to be separate so we can reinterpret some of the dict object as DataFrame objects before assuming their contents should be callable.